File: daily_quest/lib/daily_quest/domain/entity/daily_quest.dart
import 'package:equatable/equatable.dart';

import 'task.dart';

class DailyQuest extends Equatable {
  final String timestamp;
  final List<Task> tasks;

  const DailyQuest({
    required this.timestamp,
    required this.tasks,
  });

  DailyQuest addTask(Task task) {
    return DailyQuest(timestamp: timestamp, tasks: [...tasks, task]);
  }

  @override
  List<Object?> get props => [timestamp, tasks];
}

File: QuestServices/Sources/QuestServices/Models/DailyQuest.swift
//
//  DailyQuest.swift
//  DailyQuest
//
//  Created by Khoi Nguyen on 30/5/24.
//

import Foundation

public struct DailyQuest: Equatable {
    public let id: String
    public let timestamp: Date
    public let tasks: [DailyTask]

    public init(id: String, timestamp: Date, tasks: [DailyTask]) {
        self.id = id
        self.timestamp = timestamp
        self.tasks = tasks
    }
}
